pkg/controller/prpqr_reconciler/pjstatussyncer/pjstatussyncer.go

```diff
@@ -111,23 +117,49 @@ func (r *reconciler) reconcile(ctx context.Context, log *logrus.Entry, req contr
 	}
 	for i, job := range prpqr.Status.Jobs {
 		if job.ProwJob == pj.Name && !reflect.DeepEqual(pj.Status, job.Status) {
-			if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				prpqr := &v1.PullRequestPayloadQualificationRun{}
-				if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: prpqrName}, prpqr); err != nil {
-					return fmt.Errorf("failed to get the PullRequestPayloadQualificationRun: %s in namespace %s: %w", prpqrName, req.Namespace, err)
-				}
+			prpqrMutations = append(prpqrMutations, func(prpqr *v1.PullRequestPayloadQualificationRun) {
 				prpqr.Status.Jobs[i].Status = pj.Status
+			})
+		}
+	}
 
-				logger.WithField("to_state", pj.Status.State).Info("Updating PullRequestPayloadQualificationRun...")
-				if err := r.client.Update(ctx, prpqr); err != nil {
-					return err
-				}
-				return nil
-			}); err != nil {
-				return fmt.Errorf("failed to update PullRequestPayloadQualificationRun %s: %w", prpqr.Name, err)
-			}
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		prpqr := &v1.PullRequestPayloadQualificationRun{}
+		if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: prpqrName}, prpqr); err != nil {
+			return fmt.Errorf("failed to get the PullRequestPayloadQualificationRun: %s in namespace %s: %w", prpqrName, req.Namespace, err)
+		}
+
+		for _, mutate := range prpqrMutations {
+			mutate(prpqr)
 		}
+
+		if hasAllJobsFinished(prpqr.Status.Jobs) {
+			prpqr.Status.Conditions = append(prpqr.Status.Conditions, metav1.Condition{
+				Type:               conditionAllJobsFinished,
+				Status:             metav1.ConditionTrue,
+				LastTransitionTime: metav1.Time{Time: time.Now()},
+				Reason:             conditionAllJobsFinished,
+				Message:            "All jobs have finished.",
+			})
+		}
```

Comment:
The condition needs to go there even if not all jobs finished, just with `Status` set to `false` instead of true` and a different message ("X jobs are still running")

The logic is:
- Compute the condition that should be there (just type, status, reason, message)
- Compare with the condition in the status of the matching type (all members)
- If the condition is not there at all, or if the condition differs, update the condition